### PR TITLE
Leave exposure values undecorated when exposures specify `decorate: false`

### LINKED ISSUE #83: Add option to prevent exposure values from being decorated with parts
Sometimes, particularly for private exposures, you just don't want a part involved at all. We should support something like this:

```ruby
expose :foo, decorate: false do
  # ...
end
```

### PR DESCRIPTION
**Pull request by timriley**
Resolves #83

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Sorry, this is closed.

But, has it been considered to take the opposite default? I mean, convert decoration into an opt-in instead of an opt-out feature. At first sight, it seems it could give a better performance by default if we don't create decorator objects that will never be used in the case we forget the `decorate: false` flag. In the opposite case, should we forget to add `decorate: true` we'd be reminded by a runtime error. Also, I don't think that most part of exposures will be decorated in a real scenario.

**Comment by timriley** [comment]
Hi @waiting-for-dev 👋. I'm still more than happy to hear your feedback even on closed PRs! FWIW, I plan to wait until most of the "1.0" milestone issues are all done before doing a big release of them all together, so we still have some time.

Anyway, I'm quite sure we want to want to make decoration with view parts the lowest friction option here, and therefore the default. View parts are a big part of how dry-view works, and a big part of how dry-view helps you clean up your view layer. Defaulting to view part decoration, along with the part class inference in the new PartBuilder makes it easy to drop a piece of view-specific logic into the right place.

And if you think that view parts with custom logic will not be the majority case for exposures, then I'd argue that you're not (yet) using them enough ;) (Example, in the "admin" sub-app of a recent project of ours, we have 38 part classes, and nearly every entity-type we expose would have view logic on a corresponding part class).

In short, parts one of dry-view's key differentiators and I want to help view developers to embrace them :)

The reason I added `decorate: false` here is because IMO view parts may make less sense when you're exposing e.g. "primitive" values like strings or numbers, or if you're using private exposures to share some similar values among other exposures.

**Comment by waiting-for-dev** [comment]
Fair enough! Yes, I see that for entities it is really a good practice to encapsulate their view logic in view parts. Maybe my concern comes from the fact that I try to use templates as absolute dumb guys. For example, I don't even want to call `<%= Date.today.year %>` there, and I prefer the view to expose a `:year` variable. For that reason I find cumbersome having to remember to add `decorate: false` each time, and I really prefer to be explicit for features that can add a performance overhead. But, no problem, I can live with that :smile:

**Comment by solnic** [comment]
> For example, I don't even want to call <%= Date.today.year %> there, and I prefer the view to expose a :year variable

This is precisely THE way of using dry-view. It is data-driven, dumb template rendering, with as little logic in the templates as possible. A more typical scenario though, is to pass data structures there, so you'd have something like `article.year` or whatever, instead of just `year`.

> For that reason I find cumbersome having to remember to add decorate: false each time

We could consider adding a mechanism for setting this option automatically based on various criteria, but we need to be sure that it is a very, VERY common use case, otherwise it's not worth to make the library more complex.

**Comment by waiting-for-dev** [comment]
> We could consider adding a mechanism for setting this option automatically based on various criteria, but we need to be sure that it is a very, VERY common use case, otherwise it's not worth to make the library more complex.

Hmm... I think it could be quite confusing having `decorate:` to either `true` or `false` depending on some criteria... What about having a view configuration option to set `decorate:` default value? For example, in my case I think I would configure it to default to `true`, but the library could still ship with it to `false`.

**Comment by timriley** [comment]
Yep, I don't think we'll rush to add any sort of smart heuristic around what to decorate or not, nor do I think we'll add a controller-level setting for it. If you want to switch the defaults, you can at least do so by overriding the `.expose` method in your own base view controller subclass and setting `decorate: false` there.
